### DISCUSSION TITLE
Laravel 9 support released

### DISCUSSION THREAD

**Discussion by Kyon147** [osiset/laravel-shopify#1204] [General]
Hi all,

Laravel 9 is now supported in [release v17.2.0](https://github.com/osiset/laravel-shopify/releases/tag/v17.2.0) which you can use now.

As mentioned in the releases, this will be the last release which supports PHP7.x as there's some backwards compatibility blockers now trying to maintain 8.1 engine functionality while being backwards compatible with PHP7.x. 

v18.0 which will be coming soon, is going to be a minimum requirement of PHP8.0, this will have the latest changes to the basic shopify api.

**Comment by gnikyt** [comment]
> v18.0 which will be coming soon, is going to be a minimum requirement of PHP8.0

Yep. For more context, the underlying API library makes use of offset implementations, and PHP internally has changed the interfaces of these offset methods to return `mixed`, which is a new type in PHP8.0.

There is no way for us to support having `mixed` there or not depending on the PHP version, so we must move with the breaking change.

Additionally, there are minor deprecation warnings popping up while running tests for PHP8.0 in other areas (will submit a patch).

**Comment by polibou** [comment]
Awesome, thank you to everyone for all the handwork! Definitely looks like a lot of work to support so many outside platform changes lately.... Laravel 9, PHP 8, JWT and other new requirements from Shopify. :thumbsup: :clap:


Also, any word on whether non-embedded support will be coming back in v18 (issue #929)?

**Reply by Kyon147** [reply]
I personally can't see non-embed coming to v18 at least not in the early stages (@osiset can also offer up any potential plans he has, if any). There's still a lot of work required for the non-embed and it does not look like anyone in the community is actively working on it. 

As you mentioned, there's a lot of external platform support changes, bug fixes and general maintenance to also deal with at the same time to make sure the core package is healthy. 

Feel free though to tackle the non-embed if you need it for your own project and submit a PR if you can, the more help the merrier!

**Reply by gnikyt** [reply]
I'd have to go through my old branch/code. But I think there was some big blockers to supporting non-embedded. I believe it required a whole different auth flow, whole new set of redirects, different initial frontend views, etc.
